src/clr-addons/daterangepicker/directives/datepicker/datepicker.component.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  Output,
+  QueryList,
+  ViewChild,
+  ViewChildren,
+} from '@angular/core';
+import { ClrDateContainer, ClrPopoverToggleService } from '@clr/angular';
+import { Subscription } from 'rxjs';
+
+import { DayModel, NullableDayModel } from '../../models/day.model';
+import { OpenedDatepickersTrackerService } from '../../providers/opened-datepickers-tracker.service';
+
+@Component({
+  selector: 'clr-datepicker',
+  templateUrl: './datepicker.component.html',
+  styleUrls: ['./datepicker.component.scss'],
+})
+export class ClrDatepickerComponent implements AfterViewInit, OnDestroy {
+  /**
+   * Input value.
+   */
+  @Input()
+  public value: NullableDayModel;
+
+  /**
+   * Text label.
+   */
+  @Input()
+  public label = 'Date';
+
+  /**
+   * Minimum date that can be selected.
+   */
+  @Input()
+  public minDate: NullableDayModel = undefined;
+
+  /**
+   * Maximum date that can be selected.
+   */
+  @Input()
+  public maxDate: NullableDayModel = undefined;
+
+  @ViewChild('input', { static: false })
+  private input: ElementRef<any>;
+
+  @ViewChildren(ClrDateContainer)
+  private dateComponents: QueryList<ClrDateContainer>;
+
+  /**
+   * Event triggered when value changes.
+   */
+  @Output()
+  public valueChange = new EventEmitter<NullableDayModel>();
+
+  /**
+   * Disabled state.
+   */
+  protected disabled = false;
+
+  /**
+   * Get date.
+   * @returns JavaScript Date object.
+   */
+  protected get date(): Date | undefined | null {
+    //console.log('ClrDatepickerComponent.date.get', { value: this.value, date: this.value?.date });
+    return this.value?.toDate();
+  }
+
+  /**
+   * Set date.
+   */
+  protected set date(val: Date | undefined | null) {
+    //console.log('ClrDatepickerComponent.date.set', { val, date: val?.getDate() });
+    this.value = val == null ? null : new DayModel(val);
+    this.valueChange.emit(this.value);
+  }
+
+  /**
+   * Minimum date as attribute string.
+   * @returns Minimum date as attribute string.
+   */
+  protected get minDateAttr(): string | undefined | null {
+    return this.minDate?.toHTML5SpecDateString();
+  }
+
+  /**
+   * Maximum date as attribute string.
+   * @returns Maximum date as attribute string.
+   */
+  protected get maxDateAttr(): string | undefined | null {
+    return this.maxDate?.toHTML5SpecDateString();
+  }
+
+  /** List of subscriptions to later destroy. */
+  private subscriptions: Array<Subscription> = [];
+
+  public constructor(private readonly openedDatepickersTrackerService: OpenedDatepickersTrackerService) {}
+
+  public ngAfterViewInit(): void {
+    // eslint-disable-next-line dot-notation -- Need access to this private property.
+    const toggleService = this.dateComponents.first['toggleService'] as ClrPopoverToggleService;
```

Comment:
This is the only "hacky" part of the control component. 
I could not find an different/better solution. 
I doubt the `injector` is available; https://github.com/vmware-clarity/ng-clarity/blob/57d1a079df024646da9e285baaff693f708bcb08/projects/angular/src/utils/testing/helpers.spec.ts#L57

Without this logic, the two popovers (daterangepicker & one of the datepickers) would conflict with each other.

Do you got other suggestions?

Reply:
1. 🚫 Toggle service is private on `ClrDateContainer`.
2. 🚫 No `injector` available on the `ClrDateContainer`.
3. 🚫 No toggle service on `clrDate`.
4. 🚫 No access to `ClrDateContainer` from `clrDate`.
5. 🚫 Creating an new container component, which inherits from `ClrDateContainer`, also doesn't have access to private toggle service. (Besides, this would require us to define the same template.)

I'm out of ideas.

Reply:
Yes I already thought that it has to be hacky. To be honest we also have some hacks which naturally come at some time when you extend a library but want to be as integrated as possible.
Could you then please add a test which accesses the "toggleService" property the same way and checks if it's defined and of type ClrPopoperToggleService. That way we at least have a noisy breaking change and not a silent one.

Reply:
I added two different tests to confirm the service is still available and working.